packages/serverpod/lib/src/authentication/default_authentication_handler.dart

```diff
@@ -1,51 +1,12 @@
-import 'dart:io';
-
-import 'package:serverpod/src/generated/auth_key.dart';
-
 import '../server/session.dart';
 import 'authentication_info.dart';
-import 'scope.dart';
-import 'util.dart';
 
-/// The default [AuthenticationHandler], uses the auth_key table from the
-/// database to authenticate a user.
+/// The default [AuthenticationHandler], throws an [UnimplementedError].
 Future<AuthenticationInfo?> defaultAuthenticationHandler(
-    Session session, String key) async {
-  try {
-    // Get the secret and user id
-    var parts = key.split(':');
-    var keyIdStr = parts[0];
-    var keyId = int.tryParse(keyIdStr);
-    if (keyId == null) return null;
-    var secret = parts[1];
-
-    // Get the authentication key from the database
-    var tempSession = await session.serverpod.createSession(
-      enableLogging: false,
-    );
-
-    var authKey = await tempSession.db.findById<AuthKey>(keyId);
-    await tempSession.close();
-
-    if (authKey == null) return null;
-
-    // Hash the key from the user and check that it is what we expect
-    var signInSalt = session.passwords['authKeySalt'] ?? defaultAuthKeySalt;
-    var expectedHash = hashString(signInSalt, secret);
-
-    if (authKey.hash != expectedHash) return null;
-
-    // All looking bright, user is signed in
-
-    // Setup scopes
-    var scopes = <Scope>{};
-    for (var scopeName in authKey.scopeNames) {
-      scopes.add(Scope(scopeName));
-    }
-    return AuthenticationInfo(authKey.userId, scopes);
-  } catch (exception, stackTrace) {
-    stderr.writeln('Failed authentication: $exception');
-    stderr.writeln('$stackTrace');
-    return null;
-  }
+  Session session,
+  String key,
+) async {
+  throw UnimplementedError(
+    'Authentication not implemented. Set the param authenticationHandler in the Server constructor to override the default. Checkout the serverpod_auth package https://docs.serverpod.dev/concepts/authentication/setup.',
```

Comment:
Suggestion: Reflect the class where this should be defined, either `Serverpod` or `Serverpod Server` since the developer is actually instantiating an instance of `Serverpod` that is then started. 

We could also sell why the developer should check out the serverpod auth package a bit more. Or we could direct them to a page for how to implement authentication, then we could sell the auth package there?

I added two suggestions in the suggestion bellow.

```suggestion
    'Authentication not implemented. Set the param authenticationHandler in the Serverpod constructor to override the default. Authentication is easily implemented using the serverpod_auth package/Read more on how to implement authentication in Serverpod here  https://docs.serverpod.dev/concepts/authentication/setup.',
```